drivers/ethernet/eth_native_posix.c

```diff
@@ -231,55 +231,134 @@ static inline struct net_if *get_iface(struct eth_context *ctx,
 #endif
 }
 
-static int read_data(struct eth_context *ctx, int fd)
+#if defined(CONFIG_NET_VLAN)
+static struct net_pkt *prepare_vlan_pkt(struct eth_context *ctx,
+					int count, u16_t *vlan_tag, int *status)
 {
-	u16_t vlan_tag = NET_VLAN_TAG_UNSPEC;
-	struct net_if *iface;
+	struct net_eth_vlan_hdr *hdr = (struct net_eth_vlan_hdr *)(ctx->recv);
 	struct net_pkt *pkt;
-	int count;
+#if defined(CONFIG_ETH_NATIVE_POSIX_VLAN_TAG_STRIP)
+	u8_t vlan_strip_len = sizeof(struct net_eth_vlan_hdr) -
```

Comment:
You could have a macro in include/net/ethernet.h for that value since it will be reused by other drivers supporting that feature.